controllers/runnerpool_controller_test.go

```diff
@@ -310,6 +317,17 @@ var _ = Describe("RunnerPool reconciler", func() {
 		}))
 
 		// runner container spec
+		Expect(d.Spec.Template).To(MatchFields(IgnoreExtras, Fields{
+			"ObjectMeta": MatchFields(IgnoreExtras, Fields{
+				"Labels": MatchKeys(IgnoreExtras, Keys{
+					"test-label":                Equal("test"),
+					constants.RunnerOrgLabelKey: Not(Equal("should-not-be-updated")),
+				}),
+				"Annotations": MatchKeys(IgnoreExtras, Keys{
+					"test-annotation": Equal("test"),
+				}),
```

Comment:
Could you check all labels and annotations? I want to check they are merged correctly.

```suggestion
				"Labels": MatchAllKeys(Keys{
					constants.AppNameLabelKey:      Equal(constants.AppName),
					constants.AppComponentLabelKey: Equal(constants.AppComponentRunner),
					constants.AppInstanceLabelKey:  Equal(...),
					constants.RunnerOrgLabelKey:    Equal(...), // 
					constants.RunnerRepoLabelKey:   Equal(...),
					"test-label":                Equal("test"),
				}),
				"Annotations": MatchAllKeys(Keys{
					"test-annotation": Equal("test"),
				}),
```